genesyscloud/flow_loglevel/resource_genesyscloud_flow_loglevel.go

```diff
@@ -0,0 +1,158 @@
+
+1.  A getAll.... function that the CX as Code exporter will use during the process of exporting Genesys Cloud.
+2.  A create.... function that the resource will use to create a Genesys Cloud object (e.g. genesycloud_flow_logLevel)
+3.  A read.... function that looks up a single resource.
+4.  An update... function that updates a single resource.
+5.  A delete.... function that deletes a single resource.
+
+Two things to note:
+
+1.  All code in these methods should be focused on getting data in and out of Terraform.  All code that is used for interacting
+    with a Genesys API should be encapsulated into a proxy class contained within the package.
+
+2.  In general, to keep this file somewhat manageable, if you find yourself with a number of helper functions move them to a
+utils function in the package.  This will keep the code manageable and easy to work through.
+*/
+// getAllFlowLogLevels retrieves all of the flow log levels via Terraform in the Genesys Cloud and is used for the exporter
+func getAllFlowLogLevels(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	ep := getFlowLogLevelProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	flowLogLevels, apiResponse, err := ep.getAllFlowLogLevels(ctx)
+	if err != nil {
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get flow log levels: %v", err), apiResponse)
+	}
+
+	for _, flowLogLevel := range *flowLogLevels {
+		resources[*flowLogLevel.Id] = &resourceExporter.ResourceMeta{Name: *flowLogLevel.Id}
+	}
+
+	return resources, nil
+}
+
+// createFlowLogLevel is used by the flow_loglevel resource to create Genesyscloud flow_loglevel
+func createFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+	log.Printf("Creating flow log level for flow  %s", flowId)
+
+	flowLogLevelRequest := getFlowLogLevelSettingsRequestFromResourceData(d)
+	flowLogLevel, apiResponse, err := ep.createFlowLogLevel(ctx, flowId, &flowLogLevelRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to create flow log level: %s %s", err, d.Id()), apiResponse)
+	}
+
+	log.Printf("Sucessfully created flow log level for flow:  %s flowLogLevelId: %s", flowId, *flowLogLevel.Id)
+
+	d.SetId(*flowLogLevel.Id)
+	return readFlowLogLevel(ctx, d, meta)
+}
+
+// readFlowLogLevels is used by the flow_loglevel resource to read a flow log level from genesys cloud.
+func readFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+
+	log.Printf("Reading readFlowLogLevel with flowId %s", flowId)
+	if flowId == "" {
```

Comment:
You don't need this check since flow_id is required in the schema this will always exists